src/AsyncStorage.js

```diff
@@ -8,13 +8,22 @@
  * @flow
  */
 
-import merge from 'deep-assign';
+const merge = (target = {}, source = {}) => {
```

Comment:
Since this is our own implementation, could you `export` this function (and write a comment that it's being exported for tests only) and write some tests? This would proof this is working same as `deep-merge`

Reply:
I had started working on the requested changes but it looks like there is already an own, native implementation of deep merge in the code: 
* https://github.com/react-native-community/async-storage/blob/master/jest/async-storage-mock.js#L99

Can this code be moved into main file? What is the best approach in this situation?

Reply:
Good catch, I forgot about that code there 😊  But this should not be a problem in that case - the `_deepMerge` function is used in Mock function. What I meant in my request was to write tests for `merge` function only, to check if it properly merges objects together.

I guess we could try to reuse `merge` in mock itself, but I'm not sure if that's good approach - the `mock` has no external (or even internal) dependencies, and I'd like to keep it that way

Reply:
I'm a little bit afraid that I don't fully understand your test plan or your testing approach. I happy to help with fixing the initial issue and the basic tests but don't want to spend a lot of time working on the full test suite for this simple change. 

So there three options I consider:
* ditching both native implementations (in code and in mock) in favor of one of recommended replace packages (this would not require creating test suite, but this still will require avoidable dependency),
* leaving native `merge` as is and include/copy tests from the `deep-merge` package,
* closing this PR as is right now with an offer to help when someone decide to tackle this issue again in the future.

Let me know what's your opinion about that.

Reply:
I'm okay with changing the dependency to something else (more tested and used), than having own implementation, so that we save time on writing proper implementation and tests.